src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/CapiHelper.cs

```diff
@@ -1292,23 +1292,23 @@ public static int ObjToHashAlgId(Object hashAlg)
                 if (hashAlg is SHA512)
                     return CapiHelper.CALG_SHA_512;
             }
-            else if (hashAlg is Type)
```

Comment:
Technically speaking, you've changed code flow here.

If `hashAlg is HashAlgorithm`, but wasn't any of the above, you'll now also execute the `as` and nullcheck before ultimately emitting the same exception.

Since the truth table doesn't change I don't strongly object; and this is probably cleaner than floating the `hashAlgType` object up to before the initial `if`.